libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java

```diff
@@ -605,12 +605,16 @@ public void message(FIXMessage message) throws IOException {
         private void handleMsgSeqNum(FIXMessage message, FIXValue msgType, long msgSeqNum) throws IOException {
             if (msgSeqNum < rxMsgSeqNum)
                 handleTooLowMsgSeqNum(message, msgType, msgSeqNum);
-            else
+            else {
                 sendResendRequest(rxMsgSeqNum);
+                if (msgType.contentEquals(Logon)) {
+                    handleLogon(message);
+                }
```

Comment:
Should this block be before or after the `sendResendRequest()` invocation? Typically `handleLogon()` would respond with a Logon(A) message. Therefore we would now end up sending ResendRequest(2) first and only then Logon(A). However, there’s still the possibility that `handleLogon()` would kick off an asynchronous operation. In that case, even reordering these two methods would not be enough to ensure that Logon(A) is sent out first. 🤔

Reply:
I agree, the reordered way is the cleaner one. Until now I was looking at the Issue from the initiator side and did not think about the acceptor side.
In my application I have such an asynchronous operation. And I handle it with a 1 second sleep before the operation. For sure a hacky workaround, but for the moment I had no better idea.

Reply:
I thought a bit about this.

I think what we should do here in case of a Logon(A) message is to invoke `handleLogon()` and _not_ invoke `sendResendRequest()`. Then in the documentation of [`FIXConnectionStatusListener#logon()`](https://github.com/paritytrading/philadelphia/blob/main/libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnectionStatusListener.java#L69) we should note that the incoming Logon(A) message might have a MsgSeqNum(34) value that was higher than the next expected MsgSeqNum(34) value, and that in this case, the application should immediately follow the Logon(A) response with a ResendRequest(2) message. Finally, we should move the [`sendResendRequest()`](https://github.com/paritytrading/philadelphia/blob/main/libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java#L729) method from `MessageHandler` into the `FIXConnection` class to make it a bit easier for applications to do that. What do you think about this plan?

Reply:
This sounds like a good plan and it seems to be the only plan if we do not want to handle this logon/resend path in a synchronized way within the library.
I will modify the pull request :thumbsup: